### DISCUSSION TITLE
Rewrite API requests in productions

### DISCUSSION THREAD

**Discussion by mittalyashu** [vercel/next.js#22455] [Help]
I have a separate server, and adding a rewrite config in `next.config.js`

```js
module.exports = {
	async rewrites() {
		return [
			{
				source: "/api/:path*",
				destination: `${process.env.SERVER_URL}/api/:path*`,
			},
		];
	},
};
```

This is not working in production.

**Comment by michrome** [comment]
Hi @mittalyashu have you tried hard-coding your server URL into the destination string?

If that works, then you know it's something to do with `process.env.SERVER_URL`. First step might be to include`console.log(process.env.SERVER_URL)` to confirm it's being read correctly during your build step.

**Reply by mittalyashu** [reply]
> have you tried hard-coding your server URL into the destination string?

Yes.

As for logging the `process.env.SERVER_URL` it does output the value.

**Reply by michrome** [reply]
The syntax looks correct and you're picking up the environment, so that's good. I did spot the below in the documentation: do you have a file in `pages/api` that's overriding your rewrite?

> Rewrites are not able to override public files or routes in the pages directory as these have higher priority than rewrites. For example, if you have `pages/index.js` you are not able to rewrite `/` to another location unless you rename the `pages/index.js file`.

**Reply by mittalyashu** [reply]
I have API codebase separately.

**Comment by bornaradinovic** [comment]
I'm having the same issue @mittalyashu , did you ever solve it?
